dev/io.openliberty.springboot.fat30.concurrency.app/src/main/java/io/openliberty/springboot/support/version30/test/concurrency/app/ConcurrencyTasks.java

```diff
@@ -28,12 +36,32 @@ public CompletableFuture<String> task1(String message) throws Exception {
 		AppRunner.assertManagedThread(message + ": Async Task 1");
 		return CompletableFuture.completedFuture("Async Task 1 passed");
 	}
-
+	
 	@Async
 	public CompletableFuture<String> task2(String message) throws Exception {
 		System.out.println("Async Task 2: " + Thread.currentThread().getName());
 		TimeUnit.SECONDS.sleep(3);
 		AppRunner.assertManagedThread(message + ": Async Task 2");
 		return CompletableFuture.completedFuture("Async Task 2 passed");
 	}
+	
+	@Async("taskExecutor1")
+	public CompletableFuture<String> task3(String message) throws FileNotFoundException, IOException {
+			 try {
+	               InputStream inputStream = ConcurrencyTasks.class.getResourceAsStream("AsyncTask3File.txt");
+	               if (inputStream != null) {
+	                   BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
+	                   String line;
+	                   while ((line = reader.readLine()) != null) {
+	                       System.out.println("Async Task 3: File content:\n" + line);
+	                   }
+	                   reader.close();
+	               } else {
+	                   System.out.println("File not found!");
+	               }
+	           } catch (Exception e) {
+	               e.printStackTrace();
+	           }
```

Comment:
Is there a reason the test needs to read this resource?  I'm not sure what the significance of that is.  I suggest you do something similar to the other async methods.
```suggestion
	public CompletableFuture<String> task3(String message) throws Exception {
		System.out.println("Async Task 3: " + Thread.currentThread().getName());
		TimeUnit.SECONDS.sleep(1);
```